src/core/TimePicker/TimePicker.tsx

```diff
@@ -0,0 +1,376 @@
+  };
+
+  return (
+    <>
+      <TimePickerColumn
+        data={getHoursList()}
+        focusedTime={focusedTime}
+        selectedTime={selectedTime}
+        onFocusChange={onHourFocus}
+        onSelectChange={onHourClick}
+        isSame={(value, date) =>
+          isSameHour(value, date, use12Hours ? period : undefined)
+        }
+        step={hourStep}
+      />
+      {precision != 'hours' && (
+        <TimePickerColumn
+          data={[...new Array(60)].map((_, index) => index)}
+          focusedTime={focusedTime}
+          selectedTime={selectedTime}
+          onFocusChange={onMinuteFocus}
+          onSelectChange={onMinutesClick}
+          isSame={areSameMinutes}
+          step={minuteStep}
+        />
+      )}
+      {precision == 'seconds' && (
+        <TimePickerColumn
+          data={[...new Array(60)].map((_, index) => index)}
+          focusedTime={focusedTime}
+          selectedTime={selectedTime}
+          onFocusChange={onSecondFocus}
+          onSelectChange={onSecondsClick}
+          isSame={areSameSeconds}
+          step={secondStep}
+        />
+      )}
+      {use12Hours && (
+        <div className='iui-time'>
+          <ol>
+            {['AM', 'PM'].map((value) => (
+              <li
+                className={cx({
+                  'iui-selected': isSamePeriod(value, selectedTime),
+                })}
+                key={value}
+                tabIndex={isSamePeriod(value, selectedTime) ? 0 : undefined}
+                // ref={(ref) => {
+                //   scrollIntoView(ref, isSamePeriod(value, selectedTime));
+                // }}
+                onClick={() => onPeriodClick(value)}
+              >
+                {value}
+              </li>
+            ))}
+          </ol>
+        </div>
+      )}
+    </>
+  );
+};
+
+export type TimePickerColumnProps = {
+  /**
+   * Data to render in column.
+   */
+  data: number[];
+  /**
+   * Current focused time.
+   */
+  focusedTime: Date;
+  /**
+   * Current selected time.
+   */
+  selectedTime?: Date;
+  /**
+   * Callback when focus is changed.
+   */
+  onFocusChange?: (value?: number) => void;
+  /**
+   * Callback when date is changed.
+   */
+  onSelectChange?: (value: number) => void;
+  /**
+   * Compare function.
+   */
+  isSame: (value: number, date: Date | undefined) => boolean;
+  /**
+   * Step.
+   */
+  step?: number;
+};
+
+const TimePickerColumn = (props: TimePickerColumnProps): JSX.Element => {
+  const {
+    data,
+    focusedTime,
+    selectedTime,
+    onFocusChange,
+    onSelectChange,
+    isSame,
+    step = 0,
+  } = props;
+  const needFocus = React.useRef(false);
+
+  React.useEffect(() => {
+    if (needFocus.current) {
+      needFocus.current = false;
+      // onFocusChange?.(undefined);
+    }
+  });
+
+  const scrollIntoView = (ref: HTMLLIElement | null, isSame: boolean) => {
+    isSame && ref?.scrollIntoView({ block: 'nearest', inline: 'nearest' });
```

Comment:
Can you use here `needFocus` instead of passing options `scrollIntoView`? Because they are not fully supported by all browsers.

Reply:
how needFocus would help me to scroll whole thing?
i tested in other browsers (like IE), it is not that bad. Scroll goes a little (to the nearest border), but its ok.
Much worse is the visual look (css issue):
![image](https://user-images.githubusercontent.com/81580355/119822081-ef8d0f80-befb-11eb-9a81-332d91b9036f.png)